core/dbt/clients/system.py

```diff
@@ -485,7 +485,7 @@ def untar_package(
 ) -> None:
     tar_path = convert_path(tar_path)
     tar_dir_name = None
-    with tarfile.open(tar_path, 'r') as tarball:
+    with tarfile.open(tar_path, 'r:gz') as tarball:
```

Comment:
The exception we were raising was in a section where the code was figuring out the compression type.  Since we know it's `gz`, set it and hopefully bubble up a better exception.